File: src/dotnet/src/Dependabot.Job/Models/CommitOptions.cs
namespace Dependabot.Job;

/// <summary>
/// Represents commit message options for pull requests.
/// </summary>
public sealed record CommitOptions
{
  /// <summary>
  /// Gets or sets the commit message prefix.
  /// </summary>
  public string? Prefix { get; init; }

  /// <summary>
  /// Gets or sets whether to include a scope in commit messages.
  /// </summary>
  public bool? IncludeScope { get; init; }
}

File: src/ruby/lib/dependabot/job/models/commit_options.rb
# typed: strict
# frozen_string_literal: true

require 'sorbet-runtime'

module Dependabot
  module Job
    module Models
      # Represents commit message options for pull requests.
      class CommitOptions < T::Struct
        extend T::Sig

        # The commit message prefix.
        const :prefix, T.nilable(String), default: nil

        # Whether to include a scope in commit messages.
        const :include_scope, T.nilable(T::Boolean), default: nil

        sig { params(hash: T::Hash[Symbol, T.untyped]).returns(CommitOptions) }
        def self.from_hash(hash)
          new(
            prefix: hash[:prefix],
            include_scope: hash[:include_scope]
          )
        end

        sig { returns(T::Hash[Symbol, T.untyped]) }
        def to_hash
          {
            prefix: prefix,
            include_scope: include_scope
          }.compact
        end
      end
    end
  end
end
